pkg/controller/statefulset/stateful_set_control.go

```diff
@@ -269,6 +278,244 @@ func (ssc *defaultStatefulSetControl) getStatefulSetRevisions(
 	return currentRevision, updateRevision, collisionCount, nil
 }
 
+func slowStartBatch(initialBatchSize int, remaining int, fn func(int) (bool, error)) (int, error) {
```

Comment:
Nice to have: we have this code (either identical named method or identical functionality) spread through several controllers. I counted daemon set, job, replica set, and now this. In a followup, can you extract all of these into single function living in `pkg/controller/controller_utils.go`?